src/panels/config/helpers/forms/ha-input_number-form.ts

```diff
@@ -90,19 +103,79 @@ class HaInputNumberForm extends LitElement {
             "ui.dialogs.helper_settings.input_number.max"
           )}
         ></paper-input>
-        <paper-input
-          .value=${this._initial}
-          .configValue=${"initial"}
-          type="number"
-          @value-changed=${this._valueChanged}
-          .label=${this.hass!.localize(
-            "ui.dialogs.helper_settings.generic.initial_value"
-          )}
-        ></paper-input>
+        ${this.hass.userData?.showAdvanced
+          ? html`
+              <ha-paper-dropdown-menu
+                label-float
+                dynamic-align
+                .label=${this.hass.localize(
+                  "ui.dialogs.helper_settings.input_number.mode"
+                )}
+              >
+                <paper-listbox
+                  slot="dropdown-content"
+                  attr-for-selected="item-mode"
+                  .selected=${this._mode}
+                  @selected-changed=${this._modeChanged}
+                >
+                  <paper-item item-mode="slider">
+                    ${this.hass.localize(
+                      "ui.dialogs.helper_settings.input_number.slider"
+                    )}
+                  </paper-item>
+                  <paper-item item-mode="box">
+                    ${this.hass.localize(
+                      "ui.dialogs.helper_settings.input_number.box"
+                    )}
+                  </paper-item>
+                </paper-listbox>
+              </ha-paper-dropdown-menu>
+              ${this._mode === "slider"
+                ? html`
+                    <paper-input
+                      .value=${this._step}
+                      .configValue=${"step"}
+                      type="number"
+                      @value-changed=${this._valueChanged}
+                      .label=${this.hass!.localize(
+                        "ui.dialogs.helper_settings.input_number.step"
+                      )}
+                    ></paper-input>
+                  `
+                : ""}
+              <paper-input
+                .value=${this._unit_of_measurement}
+                .configValue=${"unit_of_measurement"}
+                @value-changed=${this._valueChanged}
+                .label=${this.hass!.localize(
+                  "ui.dialogs.helper_settings.input_number.unit_of_measurement"
+                )}
+              ></paper-input>
+              <br />
+              ${this.hass!.localize(
```

Comment:
This should also be only visible for advanced mode.

Reply:
It is, standard mode only shows the `min`, `max`, `icon`, `name`